src/library/scala/StringContext.scala

```diff
@@ -158,7 +195,90 @@ case class StringContext(parts: String*) {
 }
 
 object StringContext {
+  /**
+    * Linear time glob-matching implementation.
+    * Adapted from https://research.swtch.com/glob
+    *
+    * @param patternChunks The non-wildcard portions of the input pattern,
+    *                      separated by wildcards
+    * @param input The input you wish to match against
+    * @return None if there is no match, Some containing the sequence of matched
+    *         wildcard strings if there is a match 
+    */
+  def glob(patternChunks: Seq[String], input: String): Option[Seq[String]] = {
+    var patternIndex = 0
+    var inputIndex = 0
+    var nextPatternIndex = 0
+    var nextInputIndex = 0
+
+    val matchIndices = patternChunks
+      .map(_.length + 1)
+      .inits.map(_.sum - 1)
+      .toIndexedSeq
+      .reverse
+      .drop(1)
+      .dropRight(1)
```

Comment:
Any suggestions on how to write this bit better performance while still being reasonably elegant would be welcome